test/aws4.jl

```diff
@@ -0,0 +1,43 @@
+using Dates
+using Test
+using HTTP
+using HTTP: Headers, URI
+using HTTP.AWS4AuthRequest: sign_aws4!
+
+# Based on https://docs.aws.amazon.com/general/latest/gr/signature-v4-test-suite.html
+
+@testset "AWS Signature Version 4" begin
+    withenv("AWS_ACCESS_KEY_ID" => "AKIDEXAMPLE",
+            "AWS_SECRET_ACCESS_KEY" => "wJalrXUtnFEMI/K7MDENG+bPxRfiCYEXAMPLEKEY") do
```

Comment:
Would be good to add a comment mentioning these aren't real credentials. I sure hope they aren't real.